### DISCUSSION TITLE
Artillery AWS Fargate triggers SecurityHub already due to no read-only access to filesystem

### DISCUSSION THREAD

**Discussion by drissamri** [artilleryio/artillery#3420] [Artillery]
We run Artillery on AWS using the fargate option but the automated scans by AWS SecurityHub raise an alarm that _**containers should only have read-only access to its root system**_

<img width="1376" alt="Screenshot 2024-11-27 at 11 42 35" src="https://github.com/user-attachments/assets/9feddf95-efaf-4b19-87bf-15cce9166e4d">

I don't see an option to customize the Fargate deployment? Is this something that can be fixed in the framework itsefl? There is an option readonlyRootFilesystem that can be enabled on a Fargate Task which would resolve this security alert. 

If the container does need to write filesystem, there are other options to do this like using Docker volumes to get rid of this error: https://repost.aws/questions/QUZufioV63TciPZZ0onYdsdA/unable-to-limit-ecs-containers-to-read-only-access-to-root-filesystem

**Comment by hassy** [comment]
Artillery containers rely on having a writeable filesystem - workers need to be able to download test assets (and install npm dependencies if your tests use them)

**Reply by drissamri** [reply]
Hi, I just updated the post. There are alternatives to avoid root system file write access like using docker volumes for example or the /tmp directory: https://repost.aws/questions/QUZufioV63TciPZZ0onYdsdA/unable-to-limit-ecs-containers-to-read-only-access-to-root-filesystem

**Reply by hassy** [reply]
Yep definitely something we should do

**Reply by drissamri** [reply]
Should I create an issue for this? Or what's the best way to get this going? 

Can you point me to what part of the code actually create the TaskDefinition in the code? I was some references in aws-legacy, but not sure if that still used?

**Reply by hassy** [reply]
Yes, an issue would be great!

The relevant code is here:
* Task definition on Fargate: https://github.com/artilleryio/artillery/blob/main/packages/artillery/lib/platform/aws-ecs/legacy/run-cluster.js#L1114
* `sync_test_data` and `install_npm_dependencies` in the worker: https://github.com/artilleryio/artillery/blob/main/packages/artillery/lib/platform/aws-ecs/worker/loadgen-worker#L205 - possibly some other places in that code as well

**Reply by hassy** [reply]
Actually @drissamri I'll just convert this discussion into an issue
